weasyl/controllers/moderation.py

```diff
@@ -219,3 +219,187 @@ def modcontrol_copynotetostaffnotes_post_(request):
+
+    Only displays items that have the `is_spam` flag set, and are not hidden (which is used as the 'reviewed' flag).
+
+    :param request: The Pyramid request.
+    :return: A rendered page for the journal spam queue.
+    """
+    query = define.engine.execute("""
+        SELECT lo.login_name, su.submitid, su.title, su.content
+        FROM login lo
+        INNER JOIN submission su USING (userid)
+        WHERE su.is_spam IS TRUE
+            AND su.settings !~ 'h'
+    """).fetchall()
+    return Response(define.webpage(
+        request.userid,
+        "modcontrol/spamqueue/submission.html",
+        [query],
+        title="Submission Spam Queue"
+    ))
+
+
+@moderator_only
+@token_checked
+def modcontrol_spamqueue_submission_post_(request):
+    """
+    A Pyramid route controller for approving for display or items which the spam filter has tagged as being spam.
+    :param request: The Pyramid request.
+    :return/raises: HTTPSeeOther for the journal spam queue.
+    """
+    directive = request.params.get("directive")
+    if directive:
+        submitid, action = directive.split(",")
+    else:
+        raise HTTPSeeOther("/modcontrol/spamqueue/submission")
+
+    if action == "approve":
+        # Approve and insert the journal into the notifications table.
+        submitid, userid, rating, settings, content, ua_id, ip_addr = define.engine.execute("""
+            UPDATE submission
+            SET is_spam = FALSE
+            WHERE submitid = %(id)s
+            RETURNING submitid, userid, rating, settings, content, submitter_user_agent_id, submitter_ip_address
+        """, id=submitid).first()
+        # Update the spam filtering backend to indicate that this was not spam
+        spam_filtering.submit(
+            is_ham=True,
+            user_ip=ip_addr,
+            user_agent_id=ua_id,
+            user_id=userid,
+            comment_type="submission",
+            comment_content=content,
+        )
+        welcome.submission_insert(userid=userid, submitid=submitid, rating=rating, settings=settings)
+    elif action == "reject":
+        moderation.hidesubmission(submitid=submitid)
+    else:
+        raise HTTPSeeOther("/modcontrol/spamqueue/submission")
+
+    raise HTTPSeeOther("/modcontrol/spamqueue/submission")
+
+
+@moderator_only
+@token_checked
+def modcontrol_spam_remove_post_(request):
+    """
+    Submits content to the spam filtering backend, and hides it from view.
+
+    Either `submitid` or `journalid` must be present in the request's parameters.
+
+    :param request: The Pyramid request.
+    :subparam request.params['submitid']: If present, the submission's ID number.
+    :subparam request.params['journalid']: If present, the journal's ID number.
+    :return/raises: HTTPSeeOther to /modcontrol/suspenduser.
+    """
+    submitid = request.params.get('submitid')
+    journalid = request.params.get('journalid')
+    if sum([1 for item in [submitid, journalid] if item is not None]) != 1:
```

Comment:
```suggestion
    if (submitid is None) ^ (journalid is None):
```

```suggestion
    if sum(item is not None for item in [submitid, journalid]) != 1:
```

One of these options may or may not be clearer?